python/ray/train/torch.py

```diff
@@ -297,55 +405,17 @@ def prepare_data_loader(
         move_to_device (bool): If set, automatically move the data
             returned by the data loader to the correct device.
     """
+    return get_accelerator(TorchAccelerator).prepare_data_loader(
+        data_loader,
+        add_dist_sampler=add_dist_sampler,
+        move_to_device=move_to_device,
+    )
 
-    # Only add Distributed Sampler if the following conditions hold:
-    # 1. More than one training worker is being used.
-    # 2. A DistributedSampler has not already been added by the user.
-    # 3. The dataset is not an IterableDataset. Samplers do not worker with
-    # IterableDatasets.
-    if (
-        train.world_size() > 1
-        and not isinstance(data_loader.sampler, DistributedSampler)
-        and not (
-            hasattr(data_loader, "dataset")
-            and isinstance(data_loader.dataset, IterableDataset)
-        )
-        and add_dist_sampler
-    ):
-
-        def with_sampler(loader):
-            # Automatically set the DistributedSampler
-
-            # If using a sampler, the shuffle attribute in the
-            # DataLoader must be set to False.
-            # Instead the shuffling is determined by the shuffle attribute
-            # in the DistributedSampler.
-            # We identify if shuffling is enabled in the passed in
-            # DataLoader by seeing if the sampler for the DataLoader is a
-            # SequentialSampler.
-            shuffle = not isinstance(loader.sampler, SequentialSampler)
-
-            data_loader_args = {
-                "dataset": loader.dataset,
-                "batch_size": loader.batch_size,
-                "shuffle": False,
-                "num_workers": loader.num_workers,
-                "collate_fn": loader.collate_fn,
-                "pin_memory": loader.pin_memory,
-                "drop_last": loader.drop_last,
-                "timeout": loader.timeout,
-                "worker_init_fn": loader.worker_init_fn,
-                "sampler": DistributedSampler(loader.dataset, shuffle=shuffle),
-            }
-            return DataLoader(**data_loader_args)
-
-        data_loader = with_sampler(data_loader)
-
-    if move_to_device:
-        device = get_device()
-        data_loader = _WrappedDataLoader(data_loader, device)
-
-    return data_loader
+
+@PublicAPI(stability="beta")
+def accelerate() -> None:
+    """Enables training optimizations."""
+    set_accelerator(TorchAccelerator())
```

Comment:
Should we wrap this in a try-except so that we can raise a more pertinent error message if the user calls `train.torch.accelerate()` twice?

Reply:
Yeah. `RuntimeError` is a pretty broad except, but for the sake of a better error message I think it should be fine